utils/native/utils.h

```diff
@@ -0,0 +1,91 @@
+/*
```

Comment:
util folder name is too general.

I'm not sure where is the best location for jni shared code. I think we should be able to put them under api module:

api/src/main/native/shared/